### PR TITLE
#125: Preserve the original file permission by default

### PR DESCRIPTION
**Pull request by xerial**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=1.x&zinc=1.x&io=pull/126/head&librarymanagement=1.x&util=1.x&website=1.x&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Review by eed3si9n — CHANGES_REQUESTED** [review]
On hold until we figure out what's going on here.
https://github.com/sbt/io/issues/125
Likely recommendation is to just use `setPermissions` method introduced in https://github.com/sbt/io/pull/76 if that's what's needed.

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/407/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: 1.x-1.x-pull/126/head-1.x-1.x-1.x)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/407/">has checked</A> the following projects against Scala 2.12,
tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/tree/1.x">1.x                 </A> | sbt/sbt@8cfec5b3bfb415abe2e3a1d9a43611b1695f159e
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.x">1.x                 </A> | sbt/zinc@2fade533ea7a6f28c5ddd1a0f23e9a03358cab44
  io                | <A HREF="https://github.com/sbt/io/pull/126">pull/126/head       </A> | sbt/io@ddd1c1edf9c5f454b66765a76c274433956c1aa5
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/1.x">1.x                 </A> | sbt/librarymanagement@79c7483982d3225aa0efd5120f67a0289070f1af
  util              | <A HREF="https://github.com/sbt/util/tree/1.x">1.x                 </A> | sbt/util@2022713ff50d46849a4b1c93ec571287814c7055
  website           | <A HREF="https://github.com/sbt/website/tree/1.x">1.x                 </A> | 

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=1.x&zinc=1.x&io=pull/126/head&librarymanagement=1.x&util=1.x&website=1.x&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

File: io/src/main/contraband/io.contra [hunk 1]
```diff
@@ -16,4 +16,6 @@ type CopyOptions {
   ## If `true` the executable properties are copied.
   preserveExecutable: boolean! = true @since("0.0.1")
 
+  ## If `true` the file permissions are copied. Setting this `true` disregards the effect of preserveExcecutable flag
+  preservePermission: boolean! = true @since("0.0.2")
```
Comment (dwijnand) [inline io/src/main/contraband/io.contra:20]: this should be since 1.1.3 or 1.2.0 (depending where we merge it) btw

**Review by dwijnand — DISMISSED** [review]

**Comment by eed3si9n** [comment]
Closing this.

**Pull request closed without merging**